src/blockchainTx/BlockchainTxController.ts

```diff
@@ -0,0 +1,55 @@
+import {Controller, Get, Post, Body, Req, Res, Param, ValidationPipe} from '@nestjs/common';
+import { BlockchainTxService } from './BlockchainTxService';
+import {BlockchainUserDto} from './data_models/BlockchainUserDto';
+import {NewIcoDto} from './data_models/NewIcoDto';
+import {CentralizedBlockchainUserDto} from './data_models/CentralizedBlockchainUserDto';
+import {NewTokenDto} from './data_models/NewTokenDto';
+import {UpdateScreenerDto} from './data_models/UpdateScreenerDto';
+import {UserRemovalDto} from './data_models/UserRemovalDto';
+import {SwapRecoveryCentralizedDto} from './data_models/SwapRecoveryCentralizedDto';
+import {AddExtraRecoveryCentralizedDto} from './data_models/AddExtraRecoveryCentralizedDto';
+
+@Controller()
+export class BlockchainTxController {
+  constructor(private readonly appService: BlockchainTxService) {}
+
+    @Post('blockchain-user')
+    async blockchainUser(@Body() newUserDto: BlockchainUserDto) {
+        return this.appService.createUser(newUserDto);
+    }
+
+    @Post('blockchain-centralized-user')
+    async blockchainCentralizedUser(@Body() newUserDto: CentralizedBlockchainUserDto) {
+        return this.appService.createCentralizedUser(newUserDto);
+    }
+
+    @Post('token-screener')
+    async tokenScreener(@Body() svandisDataDto: NewTokenDto) {
+        return this.appService.newTokenScreener(svandisDataDto);
+    }
+
+    @Post('ico-screener')
+    async icoScreener(@Body() svandisDataDto: NewIcoDto) {
+        return this.appService.newIcoScreener(svandisDataDto);
+    }
+
+    @Post('update-svandis-data')
+    async updateSvandisData(@Body() svandisDataDto: UpdateScreenerDto) {
+        return this.appService.updateSvandisData(svandisDataDto);
+    }
+
+    @Post('remove-user')
+    async removeUser(@Body() svandisDataDto: UserRemovalDto) {
```

Comment:
This can be any user. The 2 types of users mentioned previously basically just come down to passing a recovery key in or not. If you pass no recovery address then you are a "centralized User" that Svandis must help manage account. I see how we could have seperate user and screener controllers Ruben, Would this mean in BlockchainTxModule we would change   controllers: [BlockchainTxController], to   controllers: [BlockchainUsersTxController, BlockchainScreenersTxController
],

Reply:
yes